planner/core/logical_plan_builder.go

```diff
@@ -4391,7 +4391,7 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 	}
 	b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SelectPriv, dbName.L, tableInfo.Name.L, "", authErr)
 
-	if tbl.Type().IsVirtualTable() {
+	if tbl.Type().IsVirtualTable() && tbl.Meta().Name.L != "tiflash_replica" {
```

Comment:
This change seems to be hacky. Could we just mark `tiflash_replica` to be a virtual cluster table?